### PR TITLE
Amend/#210 frontend style guide

### LINKED ISSUE #190: Remove Materializecss
We ought remove materializecss because we do not all it's potential and components.
Currently we mainly use the grid system and the typography part.

Tasks to be done:
- [ ] Remove the grid with and alternative (either [angular/flex-layout](https://github.com/angular/flex-layout) or flexgridcss)
- [ ] Remove the typography part and check why it does not work correctly with angular/material which should actually manage typography but does not
- [ ] search for other usages in the project and remove resp. replace them.

### LINKED ISSUE #207: Delete does not "unselect" on manage ui
If i have an institution selected which has years and semesters etc. and then delete it, the institution is not unselected. This means that the years and semesters of the deleted institution stay there until you reload the page.

### LINKED ISSUE #210: Frontend StyleGuide
We should change some things in the frontend to completly meet the official Angular StyleGuide.

- [x] [Core Feature Module for Services only](https://angular.io/docs/ts/latest/guide/style-guide.html#!#04-11)
- [ ] [Move logic to service](https://angular.io/docs/ts/latest/guide/style-guide.html#05-15) (especially task component)
- [x] [remove on on EventEmitters](https://angular.io/docs/ts/latest/guide/style-guide.html#05-16)
- [x] [Look at that](https://angular.io/docs/ts/latest/guide/style-guide.html#06-03)

### LINKED ISSUE #211: Attribution is on top of navigation
The new attribution prevents the nagivation from being expanded on the lower half of the button, also see the image:
image

### PR DESCRIPTION
**Pull request by M4R1KU**
This closes #210, closes #207, closes #190, closes #211, closes #221, closes #223

Summary of changes:
- ​look in the issue

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] Update CI script and configs on Jenkins (`Jenkinsfile`, `auth0.yml`, `application.yml`, etc.)

Please review @jmesserli @needToRoll  / cc: @outcobra/developers.

### CODE REVIEW

**Comment by M4R1KU** [comment]
Includes both other pullrequests 😅

**Comment by jmesserli** [comment]
The build is ok even if it does not look like it ( #225 ).

File: frontend/src/app/manage/manage.component.scss [hunk 1]
```diff
@@ -91,7 +111,7 @@ oc-app:not(.oc-mobile) {
         }
 
         .entity-desc {
-            float: left;
+            //float: left;
```
Comment (jmesserli) [inline frontend/src/app/manage/manage.component.scss]: Remove comment

File: frontend/src/app/manage/manage.component.scss [hunk 2]
```diff
@@ -48,6 +48,27 @@ oc-app:not(.oc-mobile) {
     display: flex;
     text-overflow: ellipsis;
 
+    .manage-col {
+        padding: 0 0.5rem;
+
+
+        >   ul {
```
Comment (jmesserli) [inline frontend/src/app/manage/manage.component.scss]: Remove two unnecessary spaces

File: frontend/src/app/task/task.component.ts [hunk 3]
```diff
@@ -87,14 +87,16 @@ export class TaskComponent implements OnInit, AfterViewInit {
     }
 
     private _refreshTasksWithFilter(tasks: Task[]) {
+        console.log(this);
         this._tasks = tasks;
-        this._filteredTasks = !this._filtered ? Util.cloneArray(tasks) : this._filterTasks(Util.cloneArray(tasks));
+        this._filteredTasks = this._filterTasks(Util.cloneArray(tasks));
+        console.log(this.filteredTasks);
```
Comment (jmesserli) [inline frontend/src/app/task/task.component.ts]: Remove `console.log`

File: frontend/src/app/task/task.component.ts [hunk 4]
```diff
@@ -87,14 +87,16 @@ export class TaskComponent implements OnInit, AfterViewInit {
...
     }
 
     private _refreshTasksAfterRouteChangeAndIfCacheWasDeleted() {
         this._router.events.subscribe((event) => {
             if (event instanceof NavigationEnd && !this._taskService.hasCache()) {
-                this._taskService.readAll().subscribe(this._refreshTasksWithFilter);
+                this._taskService.readAll().subscribe((tasks) => this._refreshTasksWithFilter(tasks));
```
Comment (jmesserli) [inline frontend/src/app/task/task.component.ts]: Is this code-style or compiler warning related?

File: frontend/src/index.html [hunk 5]
```diff
@@ -7,6 +7,7 @@
 
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <link rel="icon" type="image/x-icon" href="favicon.ico">
+    <link href="https://fonts.googleapis.com/css?family=Montserrat" rel="stylesheet">
```
Comment (jmesserli) [inline frontend/src/index.html]: We could also get this through npm here: https://www.npmjs.com/package/typeface-montserrat
(Maybe this is worth discussing)

File: frontend/src/sass/_styles.scss [hunk 6]
```diff
@@ -46,7 +52,7 @@ oc-app {
     margin: 0 auto;
 }
 
-.error {
+/*.error {
```
Comment (jmesserli) [inline frontend/src/sass/_styles.scss]: Remove comment

File: frontend/src/app/task/task.component.ts (hunk 4 above)
Comment (M4R1KU) [inline frontend/src/app/task/task.component.ts]: Scoping issues

**Pull request merged**